### DISCUSSION TITLE
panicked at 'Invalid route "/": insertion failed due to conflict with previously registered route: /*__private__axum_fallback'

### DISCUSSION THREAD

**Discussion by alcroito** [tokio-rs/axum#2012] [Q&A]
- [x] I have looked for existing issues (including closed) about this

### Version

```
$ cargo tree | grep axum
└── axum v0.6.18
    ├── axum-core v0.3.4
```

### Platform

macOS 12.4

### Description

Hi, I have a small project where I try to split the routers for static file serving and an api service. These are then nested under the main router.

Starting with axum 0.6.13 (including latest 0.6.18), I get a panic at runtime. 
The issue appears to be that i'm specifying a nested fallback handler for '/'.
The code didn't panic with 0.6.12.
The api_router doesn't really need to be there (the panic happens without), i just kept it for completeness to show how i structure my project.

Small repro:

```
use axum::{routing::get, Router};

fn api_router() -> Router {
    Router::new().route("/someapi", get(|| async { todo!() }))
}

fn static_files_router() -> Router {
    Router::new().fallback(get(|| async { todo!() }))
}

fn make_router() {
    let _final_router = Router::new()
        .nest("/", static_files_router())
        .nest("/api", api_router());
}

fn main() {
    make_router();
}
```

First run:

```
thread 'main' panicked at 'Invalid route "/": insertion failed due to conflict with previously registered route: /*__private__axum_fallback', src/main.rs:13:10
stack backtrace:
```

Second run of the same binary:
```
thread 'main' panicked at 'Invalid route "/*__private__axum_fallback": insertion failed due to conflict with previously registered route: /*__private__axum_fallback', src/main.rs:13:10
```

The panic seems to switch between those two messages.

**Comment by davidpdrsn** [comment]
Don't nest at "/". Use fallbacks instead.
